src/safir/testing/kubernetes.py

```
@@ -742,6 +824,426 @@ async def list_namespaced_event(
+        return ingress
+
+    # JOB API
+
+    async def create_namespaced_job(self, namespace: str, body: V1Job) -> None:
+        """Create a job object.
+
+        This will in turn create a pod object.  This pod object will have a
+        label "job-name", which will be set to the job's name.  If the mock's
+        ``initial_pod_phase`` is ``Running``, then the Job status will
+        have ``active`` set to 1.
+
+        Parameters
+        ----------
+        namespace
+            Namespace in which to create the object.
+        body
+            Object to create.
+
+        Raises
+        ------
+        kubernetes_asyncio.client.ApiException
+            Raised with 409 status if the object already exists.
+        """
+        self._maybe_error("create_namespaced_job", namespace, body)
+        self._update_metadata(body, "batch/v1", "Job", namespace)
+        name = body.metadata.name
+        self._store_object(namespace, "Job", name, body)
+        # Pretend to spawn a pod
+        # Use our metadata for spawned pod metadata if we weren't given it.
+        template = body.spec.template
+        podmd = template.metadata
+        if podmd is None:
+            podmd = V1ObjectMeta(
+                namespace=namespace,
+                labels=body.metadata.labels.copy(),
+                annotations=body.metadata.annotations.copy(),
+            )
+        podmd.name = name + "-abcde"  # Repeatable for testing
+        podmd.labels["job-name"] = name
+        await self.create_namespaced_pod(
+            namespace, V1Pod(metadata=podmd, spec=template.spec)
+        )
+        pod = await self.read_namespaced_pod(
+            name=podmd.name, namespace=namespace
+        )
+        if pod.status.phase == "Running":
+            body.status = V1JobStatus(active=1)
+
+    async def delete_namespaced_job(
+        self, name: str, namespace: str, propagation_policy: str = "Foreground"
```

If you want to force all of our code to provide the value `Foreground`, then this shouldn't have a default and you should assert that the correct value is passed in. Otherwise, the parameter shouldn't be present unless it is supported and causes a behavior change, which it doesn't look like it does. We don't want to just silently accept other values that we don't support.